src/modules/ekf2/EKF/terrain_estimator.cpp

```diff
@@ -66,6 +66,10 @@ void Ekf::runTerrainEstimator(const imuSample &imu_delayed)
 	if (!_control_status.flags.in_air) {
 		_last_on_ground_posD = _state.pos(2);
 		_control_status.flags.rng_fault = false;
+
+	} else if (!_control_status_prev.flags.in_air) {
+		// Let the estimator run freely before arming for bench testing purposes, but reset on takeoff
```

Comment:
May I ask why this reset is required in transition to takeoff?

Reply:
Because I've seen cases where a distance sensor would report a high valid data (~3m) while on ground and since since the `velocity = flow_rate * distance`, it's safer to always takeoff with a small distance estimate.
If the distance is under-estimated, the drone will drift a bit, but if it's over-estimated it can directly diverge, which is much more dangerous.

Reply:
I added a comment describing this directly in the code.